nrf-hal-common/src/uarte.rs

```diff
@@ -135,31 +201,26 @@ where
             unsafe { w.bits(1) });
 
         // Wait for transmission to end.
-        let mut endtx;
-        let mut txstopped;
-        loop {
-            endtx = self.0.events_endtx.read().bits() != 0;
-            txstopped = self.0.events_txstopped.read().bits() != 0;
-            if endtx || txstopped {
-                break;
-            }
+        while self.0.events_endtx.read().bits() == 0 {
+            // TODO: Do something here which uses less power. Like `wfi`.
         }
 
         // Conservative compiler fence to prevent optimizations that do not
         // take in to account actions by DMA. The fence has been placed here,
         // after all possible DMA actions have completed.
         compiler_fence(SeqCst);
 
-        if txstopped {
```

Comment:
Maybe remove the `Transmit` error? it'd be a breaking change though.

Reply:
From the enum? Might as well leave it in for now in case we do need it again.